kerastuner/applications/efficientnet.py

```diff
@@ -51,29 +58,34 @@ class HyperEfficientNet(hypermodel.HyperModel):
               One of `input_shape` or `input_tensor` must be
               specified.
         classes: number of classes to classify images into.
-        augmentation: optional Model or HyperModel for image augmentation.
+        augmentation_model: optional Model or HyperModel for image augmentation.
         **kwargs: Additional keyword arguments that apply to all
             HyperModels. See `kerastuner.HyperModel`.
     """
     def __init__(self,
                  input_shape=None,
                  input_tensor=None,
                  classes=None,
-                 augmentation=None,
-                 **kwargs):
+                 augmentation_model=None, **kwargs):
 
         super(HyperEfficientNet, self).__init__(**kwargs)
         if not classes:
             raise ValueError('You must specify `classes`')
 
         if input_shape is None and input_tensor is None:
             raise ValueError('You must specify either `input_shape` '
-                           'or `input_tensor`.')
+                             'or `input_tensor`.')
 
         self.input_shape = input_shape
         self.input_tensor = input_tensor
         self.classes = classes
-        self.augmentation = augmentation
+        if (isinstance(augmentation_model, hypermodel.HyperModel) or 
```

Comment:
Also, you can put all the param check in the beginning of __init__ and exist early if possible.